### PR TITLE
Add PXT_TARGET #define in codal

### PR DESCRIPTION
**Pull request by mmoskal**
This is generic version of https://github.com/microsoft/pxt-microbit/pull/3597 for future use

### CODE REVIEW

File: pxtlib/cpp.ts [hunk 1]
```diff
@@ -989,6 +989,8 @@ namespace pxt.cpp {
                 .forEach(k => optSettings["YOTTA_CFG_" + k] = optSettings[k]);
         }
 
+        optSettings["PXT_TARGET"] = JSON.stringify(appTarget.id)
```
Comment (pelikhan) [inline pxtlib/cpp.ts:992]: maybe have a optional field to override this? we have platformid already.

**Review by pelikhan — APPROVED** [review]

File: pxtlib/cpp.ts (hunk 1 above)
Comment (mmoskal) [reply pxtlib/cpp.ts:992]: I think platformid is usually "codal", so maybe not so much use here

**Comment by abchatra** [comment]
Do I need to port this? Or is this just for future?

**Comment by mmoskal** [comment]
Just for future i think

**Pull request merged**